lib/logstash/codecs/multiline.rb

```diff
@@ -114,10 +114,16 @@ class LogStash::Codecs::Multiline < LogStash::Codecs::Base
   # to events that actually have multiple lines in them.
   config :multiline_tag, :validate => :string, :default => "multiline"
 
+  # The accumulation of events can make logstash exit with an out of memory error
+  # if event boundaries are not correctly defined. This settings make sure to flush
+  # multiline events after reaching a number of lines
+  config :max_lines, :validate => :number, :default => 1000
```

Comment:
Why 1000, we should be conservative by default.. Maybe 500? if there are use cases that warrant it, users should change to 1000 or more